### ISSUE TITLE
Utils: remove `eslint-scope` types

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#4953]
`eslint-scope` has switched to use package.json exports - meaning we can no longer deep import the various bits of their codebase. The export we currently have in `utils` is to add types to the package which are augmented with our TSESTree types.

Because of this breaking change we cannot provide these types any more.
The reason I never removed `eslint-scope` from `utils` originally was laziness... People can use our `scope-manager` package instead - which is a proper fork (/ rewrite) of `eslint-scope`, rather than just a hackily typed and exported version.

### Fail

```typescript
import {TSESLintScope} from '@typescript-eslint/utils';
```

### Pass

```typescript
import * as TSESLintScope from '@typescript-eslint/scope-manager';
```

**Comment by bradzacher** [comment]
duplicate of #4041 oops
